File: 5 Kyu/5 kyu  - Perimeter of squares in a rectangle.c
typedef unsigned long long ull;
ull perimeter(int n) {

    ull prev = 1, now = 1, next = 1,sum = 0;

    for (int i = 0; i < n + 1; ++i) {
        prev = now;
        now = next;
        next = now + prev;
        sum += prev * 4;

    }

    return sum;
}

File: 5 Kyu/5 kyu - Perimeter of squares in a rectangle.rs
fn perimeter(n: u64) -> u64 {
    let mut prev;
    let mut now = 1;
    let mut nxt = 1;
    let mut sum = 0;
    
    for _i in 0..n + 1 {
        prev = now;
        now = nxt;
        nxt = now + prev;
        sum += prev * 4;
    }
    
    return sum;
}
